stores/directory/local.py

```diff
@@ -23,42 +26,81 @@ def __init__(self, local_dir: Union[str, Path]):
         assert local_dir_path.is_dir(), f"Given path {local_dir_path} is not a directory."
 
         # Store that location against the class
-
         self.local_path = local_dir_path
 
-
     def add_file(self, file: Path):
         # Abstract methods can remain without implementation for now.
         pass
 
     def has_lone_file_matching(self, pattern: str) -> bool:
-        #  To be implemented in local directory store 1 #5 
-        ...
-
-    def save_lone_file_matching(self, pattern: str, save_as: str) -> Path:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Grab a list of files matching the regex pattern to determine how many exist.
+        matching_files = [
+            f for f in self.get_file_names() if re.search(pattern, f)]
+
+        if len(matching_files) == 1:  # 1 file matched
+            return True
+        elif len(matching_files) == 0:  # 0 file matched
+            raise FileNotFoundError(f"No files found matching given pattern '{pattern}' in directory {self.local_path}")          
```

Comment:
can we return False please. 

1 file matching return True
0 files matching return False

2+ file raise an error as a duplicate of an expected file (or using too broad a pattern) will _really_ mess people up so we want stupendously loud and obvious feedback should someone manage that.

ill concede this is a little conceptually confused tbh, but the above should be serviceable for now.